pennylane/labs/resource_estimation/templates/stateprep.py

```diff
@@ -0,0 +1,595 @@
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: A dictionary containing the resource parameters:
+                * num_mps_matrices (int): the number of matrices in the MPS representation
+                * max_bond_dim (int): the bond dimension of the MPS representation
+                * precision (Union[None, float], optional): The precision used when loading the
+                  MPS matricies.
+        """
+        return {
+            "num_mps_matrices": self.num_mps_matrices,
+            "max_bond_dim": self.max_bond_dim,
+            "precision": self.precision,
+        }
+
+    @classmethod
+    def resource_rep(cls, num_mps_matrices, max_bond_dim, precision=None) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute a resource estimation.
+
+        Args:
+            num_mps_matrices (int): the number of matrices in the MPS representation
+            max_bond_dim (int): the bond dimension of the MPS representation
+            precision (Union[None, float], optional): The precision used when loading the MPS matricies.
+
+        Returns:
+            CompressedResourceOp: the operator in a compressed representation
+        """
+        params = {
+            "num_mps_matrices": num_mps_matrices,
+            "max_bond_dim": max_bond_dim,
+            "precision": precision,
+        }
+        return CompressedResourceOp(cls, params)
+
+    @classmethod
+    def default_resource_decomp(
+        cls,
+        num_mps_matrices,
+        max_bond_dim,
+        precision=None,
+        **kwargs,
+    ) -> list[GateCount]:
+        r"""Returns a list representing the resources of the operator. Each object in the list
+        represents a gate and the number of times it occurs in the circuit.
+
+        Args:
+            num_mps_matrices (int): the number of matrices in the MPS representation
+            max_bond_dim (int): the bond dimension of the MPS representation
+            precision (Union[None, float], optional): The precision used when loading
+                the MPS matricies.
+
+        Resources:
+            The resources for MPSPrep are according to the decomposition, which uses the generic
+            :class:`~.labs.resource_estimation.ResourceQubitUnitary`. The decomposition is based on
+            the routine described in `Fomichev et al. (2024) <https://arxiv.org/pdf/2310.18410>`_.
+
+        Returns:
+            list[GateCount]: A list of GateCount objects, where each object
+            represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        precision = precision or kwargs["config"]["precision_mps_prep"]
+        num_work_wires = min(
+            math.ceil(math.log2(max_bond_dim)), math.ceil(num_mps_matrices / 2)  # truncate bond dim
```

Comment:
Oh okey, we have to make sure we understand the bond dimension in the same way
I agree with this because 2^k is the bond dimension and you need k,
However, when you wrote in the example max_bond_dimension = 5, I was thinking 2^5 because it was not a power of 2. (long story short, you could change the example and write 2 ** 5 instead of 5)

Reply:
Will it always be a power of 2?